dbclient/dbclient.py

```diff
@@ -182,19 +189,30 @@ def http_req(self, http_type, endpoint, json_params, version='2.0', print_json=F
             if self.is_verbose():
                 print("{0}: {1}".format(http_type, full_endpoint))
             if json_params:
-                if http_type == 'post':
-                    if files_json:
-                        raw_results = requests.post(full_endpoint, headers=self._token,
-                                                    data=json_params, files=files_json, verify=self._verify_ssl)
-                    else:
-                        raw_results = requests.post(full_endpoint, headers=self._token,
-                                                    json=json_params, verify=self._verify_ssl)
-                if http_type == 'put':
-                    raw_results = requests.put(full_endpoint, headers=self._token,
-                                               json=json_params, verify=self._verify_ssl)
-                if http_type == 'patch':
-                    raw_results = requests.patch(full_endpoint, headers=self._token,
-                                                 json=json_params, verify=self._verify_ssl)
+                try:
```

Comment:
Can you explain me this change?

Reply:
I experienced two kind of issues when trying the migration to the dev workspace:
- ConnectionErrors during downloads (cannot open connection to x after y retries)
- Requests that hang forever (hence the added timeout parameter and the relative exception handled)
In both cases, we `continue` and retry the request